intro-Cloud/ch_Terraform.tex

```diff
@@ -0,0 +1,250 @@
+
+\strong{Note:} Make sure you download the new application credential as yaml file instead of openrc.
+
+At this point you should have a clouds.yaml text file with these lines:
+\begin{code}{}
+# This is a clouds.yaml file, which can be used by OpenStack tools as a source
+# of configuration on how to connect to a cloud. If this is your only cloud,
+# just put this file in ~/.config/openstack/clouds.yaml and tools like
+# python-openstackclient will just work with no further config. (You will need
+# to add your password to the auth section)
+# If you have more than one cloud account, add the cloud entry to the clouds
+# section of your existing file and you can refer to them by name with
+# OS_CLOUD=openstack or --os-cloud=openstack
+clouds:
+  openstack:
+    
+    auth:
+      
+      auth_url: https://cloud.vscentrum.be:13000
+      
+      application_credential_id: "xxxxxxxxxxxxxxxxxxxxxxxxx"
+      application_credential_secret: "xxxxxxxxxxxxxxxxxxxxx"
+    
+      
+        
+    region_name: "regionOne"
+        
+      
+    interface: "public"
+    identity_api_version: 3
+    auth_type: "v3applicationcredential"
+\end{code}
+
+As the file comments state, you should copy the current clouds.yaml to your VSC login node \$home
+\lstinline{login.hpc.ugent.be}: \texttt{\emph{\~\//.config/openstack/clouds.yaml}}, or locally
+if you have installed \gls{Terraform} in your own laptop or computer.
+\gls{Terraform} will use this file to authenticate to OpenStack API automatically.
+
+
+\section{Getting Terraform examples}\label{sec:getting-terraform-templ}
+
+You can connect to UGent login node \lstinline{login.hpc.ugent.be} to use terraform.
+Login to the login node with your VSC account first:
+
+\begin{prompt}
+%\shellcmd{ssh -A vscxxxxx@login.hpc.ugent.be}
+\end{prompt}
+
+If this is the first time using Terraform, download the VSC Terraform examples from github from
+\url{https://github.com/hpcugent/openstack-templates}:
+
+\begin{prompt}
+%\shellcmd{git clone https://github.com/hpcugent/openstack-templates}
+\end{prompt}
+
+Make sure you have \texttt{\emph{\~\//.config/openstack/clouds.yaml}} available from the login node (see previous section \ref{sec:app-cred-terraform}).
+
+\strong{Note:} Do not share your application credential file clouds.yaml or put this file in a public place.
+
+\begin{prompt}
+%\shellcmd{chmod 600 ~/.config/openstack/clouds.yaml}
+\end{prompt}
+
+\section{Generate Terraform template variables}\label{sec:generate-terraform-variables}
+\gls{Terraform} requires some variables to know which resources are avaiable from the cloud provider for the user or project.
+You do not have to include these variables manually, we have included a script to gather these variable IDs automatically.
+From Terraform directory cloned from git in the previous step (see: \ref{sec:getting-terraform-templ}), got to scripts directory:
```

Comment:
From the Terraform directory cloned from git in the previous step (see: \ref{sec:getting-terraform-templ}), go to the scripts directory: